shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -1209,6 +1216,12 @@ routing message to a random node so that:
                     body: ResponseBody::SaslAuthenticate(authenticate),
                     ..
                 })) => {
+                    // The broker always closes the connection after an auth failure response,
+                    // so we should do the same.
+                    if authenticate.error_code != 0 {
```

Comment:
should we do this check here or in `self.process_sasl_authenticate(authenticate)?;`?

Reply:
The only downside is having to pass in close_client_connection, but you are right, it should really be in process_sasl_authenticate.